src/datetime/tests.rs

```diff
@@ -471,7 +471,7 @@ fn test_datetime_rfc2822() {
     );
     assert_eq!(
         DateTime::parse_from_rfc2822("Wed, 18 Feb 2015 23:16:09 -0000"),
-        Ok(FixedOffset::east_opt(0).unwrap().with_ymd_and_hms(2015, 2, 18, 23, 16, 9).unwrap())
+        Ok(FixedOffset::OFFSET_UNKNOWN.with_ymd_and_hms(2015, 2, 18, 23, 16, 9).unwrap())
```

Comment:
This commit has way too much going on for me to review. Please split it into smaller parts.

Reply:
Done.